### ISSUE TITLE
Encoding issue regression for files imported into version 3.8.0

### ISSUE THREAD

**Issue by thadguidry** [OpenRefine/OpenRefine#6595]
In OpenRefine 3.8.0, extra NUL chars are being introduced between letters in a dataset, where it is not happening with previous version 3.7.9

### To Reproduce
1. First, download dataset zip from below
2. Open using OpenRefine 3.8.0
3. Notice NUL chars between letters
4. Now open using OpenRefine 3.7.9
5. Notice no NUL chars between letters

### Current Results
NUL chars are being introduced erroneously between letters.  Appears to be some kind of errant decoding/encoding issue?

### Expected Behavior
Should work as it did before with OpenRefine 3.7.9 , without the extra NUL chars being introduced.

### Screenshots

With OpenRefine 3.8.0
![image](https://github.com/OpenRefine/OpenRefine/assets/986438/1408d4db-a892-487a-b3b6-398e6b3f4414)
With OpenRefine 3.7.9
![image](https://github.com/OpenRefine/OpenRefine/assets/986438/e09c41ba-434a-497b-abd4-b4ad021cbd22)

### Versions
 - Operating System: Mac OS and Windows 11
 - Browser Version: Edge and Safari
 - JRE or JDK Version: embedded
 - OpenRefine: 3.8.0

### Datasets

Download the ZIP file from https://www.cssf.lu/en/Document/identifiers-of-aifms/
direct link: https://www.cssf.lu/wp-content/uploads/IDENTIFIANTS_AIFM.zip

### Additional context
Opening the ZIP file directly in OpenRefine or extracting it first and then opening in OpenRefine, causes the same encoding issue and appearance of NUL char between letters.

**Comment by tfmorris** [comment]
This happens on both macOS and Windows? I'm unable to reproduce it on macOS 14.4.1 (Sonoma) with either 3.8.0 or the current development head.

Did it guess UTF-16LE or did you specify it? (It's correct, but I'm curious because my system is guessing Windows-1252)

If you try switching to a different encoding (e.g. UTF-8) and then back to UTF-16LE, do the symptoms change?

It looks like it might be using the system default encoding instead of the correctly guessed encoding, but I'll have to dig further to understand where things are going awry.

**Comment by tfmorris** [comment]
I just double checked with the binary 3.8.0 kit and bundled JVM (OpenJDK 64-Bit Server VM version 11.0.23+9) to be sure it wasn't some kind of binary kit build issue and got the same results.

**Comment by thadguidry** [comment]
The "same" results of reproducing the issue? or not reproducing the issue?

**Comment by thadguidry** [comment]
I've been helping Mike Gordon on this issue, emailing back and forth and finally discovering the real issue coming up in regards to https://forum.openrefine.org/t/trouble-with-sql-export/1485

**Comment by tfmorris** [comment]
> The "same" results of reproducing the issue? or not reproducing the issue?

None of my three tests reproduced the issue on macOS.

The bug report says:

> Operating System: Mac OS and Windows 11

Did you actually test on both? What version of macOS? Which operating system are the screenshots above from?

**Comment by thadguidry** [comment]
I am testing on Windows 11.
Mike Gordon in the forum is on a MacBook Pro mid 2012 on macOS Catalina running OpenRefine 3.8.

**Comment by tfmorris** [comment]
Also not reproducible with the binary 3.8.0 kit on macOS Big Sur 11.7.10 (20G1427). I'll see if I can find a Windows machine.

**Comment by tfmorris** [comment]
Actually, it looks like my browsers are rendering things differently than yours and there are actually NULs in the data, so let me dig in more now that I can reproduce it.

**Comment by thadguidry** [comment]
Ah, in the data, just reading the bytes directly out of the compressed file.. I also see NULLs or rather `\u0000` or `char(0)`.
For example, I see the gaps within all the char(78) which is "N" where it should be `NNNNNNNN`.

```python
import zipfile

zip_file_path = 'F:\Downloads\IDENTIFIANTS_AIFM.zip'
# read a file from a windows path
with zipfile.ZipFile(zip_file_path, 'r') as myzip:
    file_name = 'IDENTIFIANTS_AIFM.csv'
    with myzip.open(file_name, 'r') as f:
        # read the first 32 bytes of the file and print the Unicode decimal codepoints
        f_list = [ord(i) for i in f.read(32).decode('unicode_escape')]
        print(f_list)
```

![image](https://github.com/OpenRefine/OpenRefine/assets/986438/67c4d36e-34ab-44e5-907d-d75d3f2dfa98)

**Comment by tfmorris** [comment]
There are a number of different problems here:
1. The fix (#6025) for the wonky Microsoft UTF-8 + BOM problem (#1241) messed up the encoding guessing for other BOM-based encoding (including the UTF-16LE used in this file), because all BOMs were being stripped, making them unavailable for the character encoding guesser.
2. The manual character encoding override is being ignored. This might have been broken for a long time, but wasn't critical as long as the encoding guessing was working correctly.
3. There are no tests for encodings with BOM, other than the recently added UTF-8 + BOM test, so there was no test coverage for this failure.

I have fixes for all of these that I'll put up in a PR.

**Comment by thadguidry** [comment]
Looks like in 3.9-SNAPSHOT that I built locally... the autodetection for the CSV file we're testing with in original comment, is not populating with `x-UTF-16LE BOM` but instead with just `UTF-16LE` when running on my Windows 11.

I'll open a new issue.
